optaplanner-examples/src/main/java/org/optaplanner/examples/taskassigning/solver/score/TaskAssigningConstraintProvider.java

```diff
@@ -0,0 +1,80 @@
+
+    private static final int BENDABLE_SCORE_HARD_LEVELS_SIZE = 1;
+    private static final int BENDABLE_SCORE_SOFT_LEVELS_SIZE = 4;
+
+    private static UniConstraintStream<Task> getInitializedTaskUniStream(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(Task.class)
+                .filter(task -> task.getEmployee() != null);
+    }
+
+    private static Constraint noMissingSkills(ConstraintFactory constraintFactory) {
+        return getInitializedTaskUniStream(constraintFactory)
+                .filter(task -> task.getMissingSkillCount() > 0)
+                .penalize("No missing skills",
+                        BendableScore.ofHard(BENDABLE_SCORE_HARD_LEVELS_SIZE, BENDABLE_SCORE_SOFT_LEVELS_SIZE, 0, 1),
+                        Task::getMissingSkillCount);
+    }
+
+    private static Constraint criticalPriorityBasedTaskEndTime(ConstraintFactory constraintFactory) {
+        return getInitializedTaskUniStream(constraintFactory)
+                .filter(task -> task.getPriority() == Priority.CRITICAL)
+                .penalize("Critical priority task end time",
+                        BendableScore.ofSoft(BENDABLE_SCORE_HARD_LEVELS_SIZE, BENDABLE_SCORE_SOFT_LEVELS_SIZE, 0, 1),
+                        Task::getEndTime);
+    }
+
+    private static Constraint minimizeMakespan(ConstraintFactory constraintFactory) {
+        return getInitializedTaskUniStream(constraintFactory)
+                .filter(task -> task.getNextTask() == null)
+                .penalize("Minimize makespan, latest ending employee first",
+                        BendableScore.ofSoft(BENDABLE_SCORE_HARD_LEVELS_SIZE, BENDABLE_SCORE_SOFT_LEVELS_SIZE, 1, 1),
+                        task -> task.getEndTime() * task.getEndTime());
+        /*
+         * TODO ^^^ Some constraints need exponential weighting. perhaps we should have dynamic
+         *  constraint weights, inferred from the entity?
+         */
+    }
+
+    private static Constraint majorPriorityTaskEndTime(ConstraintFactory constraintFactory) {
+        return getInitializedTaskUniStream(constraintFactory)
+                .filter(task -> task.getPriority() == Priority.MAJOR)
+                .penalize("Major priority task end time",
+                        BendableScore.ofSoft(BENDABLE_SCORE_HARD_LEVELS_SIZE, BENDABLE_SCORE_SOFT_LEVELS_SIZE, 2, 1),
+                        Task::getEndTime);
+    }
+
+    private static Constraint minorPriorityTaskEndTime(ConstraintFactory constraintFactory) {
+        return getInitializedTaskUniStream(constraintFactory)
+                .filter(task -> task.getPriority() == Priority.MINOR)
+                .penalize("Minor priority task end time",
+                        BendableScore.ofSoft(BENDABLE_SCORE_HARD_LEVELS_SIZE, BENDABLE_SCORE_SOFT_LEVELS_SIZE, 3, 1),
+                        Task::getEndTime);
+    }
+
+    @Override
+    public Constraint[] defineConstraints(ConstraintFactory constraintFactory) {
```

Comment:
Can we put this method first like in the other Providers? (Or move it last in all other providers - but I prefer having it first, it's a good TOC)

Reply:
@ge0ffrey I agree, but don't we have something about this in code style? It's at the end, because that's where IDE auto-placed it.

(The other methods are static, and therefore go above.)

Reply:
The other methods shouldn't be static. They're not in the other examples.
Either we make them static in all examples, or not at all.

There are very-long idea's to allow the dependency inject instances of ConstraintProvider maybe some day, so let's not make them static.